RevenueCatUI/Templates/V2/Components/Image/ImageComponentView.swift

```diff
@@ -45,21 +45,27 @@ struct ImageComponentView: View {
                 package: self.packageContext.package
             )
         ) { style in
-            RemoteImage(
-                url: style.url,
-                lowResUrl: style.lowResUrl,
-                darkUrl: style.darkUrl,
-                darkLowResUrl: style.darkLowResUrl
-            ) { (image, size) in
-                self.renderImage(image, size, with: style)
+            Group {
```

Comment:
I see this `Group` being needed all over the place. Wdyt about creating a custom view that receives the visibility boolean + the swift UI view and is in charge of displaying according to the visibility parameter? Mostly so we don't need both the `Group` + the if-else everywhere.

Reply:
Even easier, I think, in order to remove the `Group`s, would be to add the `@ViewBuilder` attribute to the `apply` parameters in the view models. E.g., in `IconComponentViewModel.swift`:
```
@ViewBuilder
    func styles(
        state: ComponentViewState,
        condition: ScreenCondition,
        isEligibleForIntroOffer: Bool,
        @ViewBuilder apply: @escaping (IconComponentStyle) -> some View
    )
```

I think it's the same as we want, and the only difference is that without the `@ViewBuilder` attribute we're forced to return 1 `some View`, whereas `@ViewBuilder` allows to provide multiple views (the case of the `if-else` statement).

Reply:
In fact, with the `@ViewBuilder` attribute, we can even remove the `else` case (removing `EmptyView()`). But that might have other consequences (I believe returning `EmptyView()` in the `else` does not have the same effect as not implementing the `else`).